apps/app/src/views/routes/Orders.tsx

```diff
@@ -94,10 +94,19 @@ const renderRow = (order: any) => {
   const { id, pharmacy, patient } = order;
   const extId = order.externalId || <Text as="i">None</Text>;
 
-  const fills = order.fills.reduce((prev: string, cur: any) => {
-    const fill = cur.treatment.name;
-    return prev ? `${prev}, ${fill}` : fill;
-  }, '');
+  const fills = order.fills
+    .reduce((prev: any[], cur: any) => {
+      // Remove duplicates based on prescription id
+      const exists = prev.some((fill) => fill.prescription.id === cur.prescription.id);
+      if (!exists) {
+        prev.push(cur);
+      }
+      return prev;
+    }, [])
+    .reduce((prev: string, cur: any) => {
+      const fill = cur.treatment.name;
+      return prev ? `${prev}, ${fill}` : fill;
+    }, '');
```

Comment:
Could we not just do like
```
// Create a set of names
const medicationNames = new Set();
// Add names to set
order.fills.map(item => medicationNames.add(item.treatment.name))
// Turn into array
const names = Array.from(medicationNames)
```
Or something? Seems like the reduce's create a lot of unnecessary looping